frontend/src/pages/LogInPage/useForm.js

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import API from "../../API";
+
+const useForm = (validate) => {
+    const [values, setValues] = useState({
+      username: "",
+      password: "",
+    });
+    
+    const [errors, setErrors] = useState({});
+    const [loginSuccess, setLoginSuccess] = useState(false);
+
+    const handleChange = e => {
+        const { name, value } = e.target;
+        setValues({
+          ...values,
+          [name]: value
+        });
+      };
+    
+
+    const clickLogin = (e) => {
+        e.preventDefault();
+
+        const valuesFormData={
+            username: values.username,
+            password: values.password
+        };
+
+        API.post("/user/login", valuesFormData)
+        .then((response) => {
+            console.log(response.data);
+            localStorage.setItem('token', API.get("/token/auth", response));
+            setLoginSuccess(true);
```

Comment:
```suggestion
            localStorage.setItem('token', `knox ${response.data.token}`);
            API.defaults.headers.Authorization = `knox ${response.data.token}`;
            navigate('/')
```
remove console log and unnecessary extra steps
the token should be returned as well after authenticating right so you dont need to fetch the token again. Also append knox so you wont add condition in you axios base. less code

Lastly, override the authorization of your axios base so you dont need to refresh to be able to get the authorization

Reply:
Good day sir, I tried the navigate in the homepage but it would not work, when I log in the page would still stay and would not redirect to the homepage so I used reload. Is it alright sir?